pkg/logs/agent.go

```diff
@@ -146,7 +156,13 @@ func NewServerless(sources *config.LogSources, services *service.Services, proce
 // Start starts all the elements of the data pipeline
 // in the right order to prevent data loss
 func (a *Agent) Start() {
-	starter := restart.NewStarter(a.destinationsCtx, a.auditor, a.pipelineProvider, a.diagnosticMessageReceiver)
+	starter := restart.NewStarter(
+		a.destinationsCtx,
+		a.auditor,
+		a.pipelineProvider,
+		a.diagnosticMessageReceiver,
+		a.schedulers,
```

Comment:
Should schedulers be started after the inputs, since they are stopped first?

Reply:
Yeah, I need to figure that out -- a complicating factor is that AD starts _after_ the logs agent.

Reply:
(I've figured this out now..)